### PR TITLE
Add visible property to table columns

### LINKED ISSUE #11423: [FEATURE] `visible` property for `TableColumns`
**Is your feature request related to a problem? Please describe.**

I am trying to use a CheckboxButtonGroup widget to toggle the visibility of columns in a DataTable. My approach to this is to use some CustomJS that adds or removes columns from the table in question upon clicking the button:

```python
# example.py

from datetime import date
from random import randint

from bokeh.io import show
from bokeh.layouts import column
from bokeh.models import (
        ColumnDataSource,
        DataTable,
        DateFormatter,
        TableColumn,
        CheckboxButtonGroup,
        CustomJS
    )
from bokeh.plotting import curdoc

data = dict(
        dates=[date(2014, 3, i+1) for i in range(10)],
        downloads=[randint(0, 100) for i in range(10)],
    )
source = ColumnDataSource(data)

columns = [
        TableColumn(field="dates", title="Date", formatter=DateFormatter()),
        TableColumn(field="downloads", title="Downloads"),
    ]
data_table = DataTable(source=source, columns=columns, width=400, height=280)

column_choice = CheckboxButtonGroup(
    labels=[
        "Date",
        "Downloads"
    ],
    active=[0, 1],
)
column_choice.js_on_click(
    CustomJS(
        args=dict(
            table=data_table, columns=columns
        ),
        code="""
            var visible_columns = []
            for (var i = 0; i < this.active.length; i++) {
                visible_columns.push(columns[this.active[i]])
            }
            table.columns = visible_columns;
        """,
    )
)

curdoc().add_root(column(column_choice, data_table))
```

```
$ bokeh serve --show example.py
```

This works fine, and the table responds as expected. However, I get serialization errors in console every time I click the button - I reported this bug in #11422.

**Describe the solution you'd like**

An alternative method to toggle column visibility would be to have a `visible` property on `TableColumns`, similar to the one for `DataTables`. This would allow me to toggle the visibility of the columns using this property, which would be simpler than what I am currently doing and would potentially circumvent the serialization errors I encountered in #11422.

**Describe alternatives you've considered**
An alternative suggested by @bryevdv to try and avoid the serialization errors was to use a hidden `DataTable` in the document as a "bank" for the columns I want to toggle on and off for the main table. I tried this approach out but encountered the same serialization errors.

**Additional context**
This issue came up while working on dask/distributed#4614.

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #11423
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by bryevdv** [comment]
@mattpap Do you have any suggestions for testing this?

**Comment by mattpap** [comment]
Like this:
https://github.com/bokeh/bokeh/blob/ac69e52b7c8ff4f817e3addcaf822e5ac81f12ab/bokehjs/test/integration/regressions.ts#L48-L61
i.e. create a table, await display, modify visibility and await ready; just under widgets, not regressions.

**Comment by bryevdv** [comment]
Just reiterating that the current process for adding the necessary artifact is quite fraught. This branch was only a ~week old but there were new tests on `branch-2.4` that showed up in the downloaded report.  Just unzipping the report added the artifacts for those newer tests that showed up as new files to add. A rebase failed while the files existed, and deleting them is difficult due to the special characters in the filenames. Eventually resorted to `fit clean -xfd` in the tests dir to be able to rebase and re-unzip. I can figure all this out but I think it is a very high bar for ay new contributor.  cc @tcmetzger for visibility in to these possibilities.

**Comment by mattpap** [comment]
Given that effectively you're not testing this branch, but a merge, then appearance of files from that merge shouldn't be surprising. In fact it can go much worse that that. A week old PR is actually quite old given there were 19 PRs merged during this timespan. I rebase my PRs pretty much every time I push (assuming there are any changes to pull in). This not only helps me to avoid major conflicts (especially after PRs like #11503), but also makes sure that I'm fully aware what I'm testing. I doubt there is much we can do to improve this, as long as we test merges, that wouldn't be prone to other issues.

**Comment by mattpap** [comment]
> (...) deleting them is difficult due to the special characters in the filenames (...)

Which exactly? Troublesome (to a shell) characters are encoded (or dropped) to make it easier to operate on baseline files.

File: examples/reference/models/data_table_server.py [hunk 1]
```diff
@@ -17,17 +17,25 @@
...
     TableColumn(field="x", title="x"),
-   TableColumn(field="y", title="y")
+    TableColumn(field="y", title="y")
 ]
 data_table = DataTable(
     source=source,
     columns=columns,
     index_position=None,
-    width=80,
+    # width=80,
```
Comment (mattpap) [inline examples/reference/models/data_table_server.py:26]: Is there a reason to keep this?

Comment (bryevdv) [inline examples/reference/models/data_table_server.py:26]:
```suggestion
```

**Pull request merged**